### PR TITLE
Accept arrays for filter options

### LINKED ISSUE #172: ignore/only{Fs} should accept arrays
```js
getWorkspaces({
  cwd,
  ignoreFs: [
    'packages/css-packs/*',
    'build/*',
    'website'
  ]
});
```

https://github.com/boltpkg/bolt/blob/b2f52a25311df2152794002f1d8792fadebe8360/src/functions/getWorkspaces.js#L7-L10

via @jamiebuilds

### PR DESCRIPTION
**Pull request by sidoshi**
Fixes #172

### CODE REVIEW

File: src/utils/options.js [hunk 1]
```diff
@@ -35,10 +35,14 @@ export function number(val: mixed, name: string): number | void {
...
-  if (flags.ignore) filterOpts.ignore = string(flags.ignore, 'ignore');
-  if (flags.ignoreFs) filterOpts.ignoreFs = string(flags.ignoreFs, 'ignoreFs');
+  let split = (s: string | void): Array<string> =>
+    typeof s === 'string' ? s.split(',').filter(Boolean) : [];
+
+  if (flags.only) filterOpts.only = split(string(flags.only, 'only'));
+  if (flags.onlyFs) filterOpts.onlyFs = split(string(flags.onlyFs, 'onlyFs'));
+  if (flags.ignore) filterOpts.ignore = split(string(flags.ignore, 'ignore'));
+  if (flags.ignoreFs)
+    filterOpts.ignoreFs = split(string(flags.ignoreFs, 'ignoreFs'));
```
Comment (ajaymathur) [inline src/utils/options.js]: the string function check if `typeof val !== string`, is should fail in case of arrays?

**Comment by sidoshi** [comment]
I don't think we will be receiving array here as these  options are received from the client args

**Comment by ajaymathur** [comment]
Thanks for clarifying, LTGM 🎉

File: src/__tests__/Project.test.js [hunk 2]
```diff
@@ -156,16 +156,16 @@ describe('Project', () => {
     let project = await Project.init(f.find('nested-workspaces'));
     let packages = await project.getPackages();
     let filtered = await project.filterPackages(packages, {
-      only: 'foo'
+      only: ['foo', 'bar']
```
Comment (lukebatchelor) [inline src/__tests__/Project.test.js]: In updating these tests you're also changing the code paths they used to test. 

Would it be possible to keep them as close to the old tests as possible and add new tests for the new functionality?

i.e

```
let filtered = await project.filterPackages(packages, {
-  only: 'foo'
+  only: ['foo']
```

That way we know the base cases still work?

**Comment by lukebatchelor** [comment]
Hiya, thanks for the contribution.

My only thought is that by not accepting strings as well we need to make this a breaking change. I think keeping it backwards compatible would be preferred.

File: src/utils/__tests__/options.test.js [hunk 3]
```diff
@@ -29,11 +29,17 @@ describe('options', () => {
   });
 
   test('toFilerOptions', () => {
-    expect(toFilterOpts({ only: 'only' })).toEqual({ only: 'only' });
-    expect(toFilterOpts({ onlyFs: 'onlyFs' })).toEqual({ onlyFs: 'onlyFs' });
-    expect(toFilterOpts({ ignore: 'ignore' })).toEqual({ ignore: 'ignore' });
-    expect(toFilterOpts({ ignoreFs: 'ignoreFs' })).toEqual({
-      ignoreFs: 'ignoreFs'
+    expect(toFilterOpts({ only: 'onlyOne,onlyTwo' })).toEqual({
```
Comment (lukebatchelor) [inline src/utils/__tests__/options.test.js]: Same here, could these be extra tests instead of replacing the old ones?

**Comment by sidoshi** [comment]
Yes, it would be better to keep this backward compatible. 

I thought it would only be a breaking change internally as there would be no breaking change for users using the CLI. But I didn't realize we expose that function from library API as well. 

I will work on making this backward compatible and also updating the tests over the weekend.

**Comment by sidoshi** [comment]
I have added string support to make this change backwards compatible

**Pull request closed without merging**